### DISCUSSION TITLE
Error handling

### DISCUSSION THREAD

**Discussion by oranichu** [backube/volsync#109] [Q&A] [answered]
Hey again :) 

I'm going through the code and was wondering what can VolSync do in case of error handling while replication.
Let's say we have a connectivity issue whilst replication of big data from one PV to another, does in have the capability to continue from where it stopped or a new replication needs to be issued ?

Maybe you can pinpoint me to the code that handles this stuff?

Thanks :)

**Comment by JohnStrunk** [comment, marked as answer]
During a sync iteration, the controller code will restart the mover pod if it doesn't complete successfully.
How retries are treated is a function of the specific mover. All three of the movers should be pretty good about resuming where they left off. The partial data that was transferred will still be in the destination's temporary volume, so the retry will just be a sync where the destination already has some of the changes, resulting in less data transfer.

In the code, you can search for the `ensureJob` functions that implement the controller-level retry (this would happen after kube's Job controller tries to restart the Pod a few times). The destination side won't take a snapshot after a failed transfer; it'll just try again.
Beyond that, there's no specific retry code because the movers don't need to do anything differently.

**Reply by oranichu** [reply]
Thanks for the answers @JohnStrunk!
